Framework/Algorithms/src/SampleCorrections/DetectorGridDefinition.cpp

```diff
@@ -0,0 +1,98 @@
+#include "MantidAlgorithms/SampleCorrections/DetectorGridDefinition.h"
+
+#include <cmath>
+
+namespace Mantid {
+namespace Algorithms {
+
+/** Initializes a DetectorGridDefinition object.
+ *  @param minLatitude Start of the latitude range.
+ *  @param maxLatitude End of the latitude range.
+ *  @param latitudePoints Number of rows.
+ *  @param minLongitude Start of the longitude range.
+ *  @param maxLongitude End of the longitude range.
+ *  @param longitudePoints Number of columns.
+ */
+DetectorGridDefinition::DetectorGridDefinition(const double minLatitude,
+                                               const double maxLatitude,
+                                               const size_t latitudePoints,
+                                               const double minLongitude,
+                                               const double maxLongitude,
+                                               const size_t longitudePoints)
+    : m_minLatitude(minLatitude), m_maxLatitude(maxLatitude),
+      m_latitudePoints(latitudePoints), m_minLongitude(minLongitude),
+      m_maxLongitude(maxLongitude), m_longitudePoints(longitudePoints) {
+  // The angular ranges might be zero in some cases preventing
+  // the spawning of a real grid. We want to avoid this.
+  const double tiny = 1e-5;
+  const double smallShift = M_PI / 300.0;
+  if (std::abs(m_minLatitude - m_maxLatitude) < tiny) {
+    m_minLatitude -= smallShift;
+    m_maxLatitude += smallShift;
+  }
+  if (std::abs(m_minLongitude - m_maxLongitude) < tiny) {
+    m_minLongitude -= smallShift;
+    m_maxLongitude += smallShift;
+  }
+  m_latitudeStep = (m_maxLatitude - m_minLatitude) /
+                   static_cast<double>(m_latitudePoints - 1);
```

Comment:
In principle, the validators in `MonteCarloAbsorption::init()` prevent this, but I guess it doesn't hurt to be extra careful here.